scripts/Tools/xmlchange

```diff
@@ -24,64 +60,72 @@ logger = logging.getLogger("xmlchange")
-                        help="the value to set")
+    parser.add_argument("--id", "-id",
+                        help="The variable to set.\n"
+                        "(Used in the alternative --id var --val value form, rather than\n"
+                        "the recommended var=value form.)")
 
-    parser.add_argument("-delimiter","--delimiter", type=str, default="," ,
-                        help="set delimiter string, default is ,")
+    parser.add_argument("--val","-val",
+                        help="The value to set.\n"
+                        "(Used in the alternative --id var --val value form, rather than\n"
+                        "the recommended var=value form.)")
 
-    parser.add_argument("-dryrun","--dryrun", type=bool , default=False,
-                        help="parse settings and print key value pairs only")
+    parser.add_argument("--file", "-file",
+                        help="XML file to edit.\n"
+                        "Generally not needed, but can be specified to ensure that only the\n"
+                        "expected file is being changed. (If a variable is not found in this file,\n"
+                        "an error will be generated.)")
 
-    parser.add_argument("-noecho", "--noecho", action="store_true",
-                        help="do not update CaseStatus with this change")
+    parser.add_argument("--delimiter","-delimiter", type=str, default="," ,
+                        help="Delimiter string in listofvalues.\n"
+                        "Default is ','.")
 
-    parser.add_argument("-append","--append", action="store_true",
-                        help="append to the existing value")
+    parser.add_argument("--dryrun","-dryrun", action="store_true",
+                        help="Parse settings and print key-value pairs, but don't actually change anything.")
 
-    parser.add_argument("-subgroup","--subgroup",
-                        help="apply to this subgroup only")
+    parser.add_argument("--noecho", "-noecho", action="store_true",
+                        help="Do not update CaseStatus with this change.\n"
+                        "This option is mainly meant to be used by cime scripts: the 'paper trail' in\n"
+                        "CaseStatus is meant to show changes made by the user, so we generally don't\n"
+                        "want this to be contaminated by changes made automatically by cime scripts.")
 
     parser.add_argument("-f","--force", action="store_true",
-                        help="ignore typing checks and store value")
+                        help="Ignore typing checks and store value.")
+
+    parser.add_argument("-loglevel",
+                        help="Ignored, only for backwards compatibility.")
 
     args = CIME.utils.parse_args_and_handle_standard_logging_options(args, parser)
 
     listofsettings = []
     if( len(args.listofsettings )):
+        expect(args.id is None, "Cannot specify both listofsettings and --id")
```

Comment:
Most of the changes in this PR just involve documentation, but the next few lines involve some new logic to do error-checking on arguments. It would be great if someone could double check this.